domain/modelagent/service/service.go

```diff
@@ -41,6 +43,19 @@ func (s *Service) GetModelAgentVersion(ctx context.Context, modelID model.UUID)
 	return s.st.GetModelAgentVersion(ctx, modelID)
 }
 
+// SetModelAgentVersion sets the agent version for the specified model.
+// The following errors can be returned:
+//   - [errors.NotValid] if the model ID is not valid.
+//
+// NB: this method was only implemented for testing purposes. You should be
+// very careful and think twice before using this in production
+func (s *Service) SetModelAgentVersion(ctx context.Context, modelID model.UUID, ver version.Number) error {
```

Comment:
Don't do this. It's only used in one test, so get the transaction running from the suite, and execute some SQL to do it.

Reply:
We need it for the jujud-controller agent tests too. That's really the only reason we implemented it.
https://github.com/barrettj12/juju/blob/1f13d1259d75c5a523a24b8457ef6bdae49f3998/cmd/jujud-controller/agent/machine_test.go#L268

Reply:
Put in test code as I've described.

Reply:
done